src/workshed.ts

```diff
@@ -0,0 +1,390 @@
+  TrainsetPiece.RANDOM_BOOZE,
+  TrainsetPiece.MYS_STATS,
+  TrainsetPiece.MUS_STATS,
+  TrainsetPiece.BUFF_FOOD_DROP,
+  TrainsetPiece.DROP_LAST_FOOD_OR_RANDOM,
+  TrainsetPiece.ORE,
+];
+
+function getPieceId(piece: TrainsetPiece): number {
+  return Math.max(0, pieces.indexOf(piece));
+}
+
+function getTrainsetPositionsUntilConfigurable(): number {
+  const pos = toInt(get("trainsetPosition", ""));
+  const configured = toInt(get("lastTrainsetConfiguration", ""));
+  const turnsSinceConfigured = pos - configured;
+
+  return Math.max(0, 40 - turnsSinceConfigured);
+}
+
+function isTrainsetConfigurable(): boolean {
+  // eslint-disable-next-line libram/verify-constants
+  return getWorkshed() === $item`model train set` && getTrainsetPositionsUntilConfigurable() <= 0;
+}
+
+function setTrainsetConfiguration(pieces: TrainsetPiece[]) {
+  visitUrl("campground.php?action=workshed");
+
+  const pieceIds = pieces
+    .map((p) => getPieceId(p))
+    .map((pieceId, index) => `slot[${index}]=${pieceId}`);
+
+  const url = `choice.php?forceoption=0&whichchoice=1485&option=1&pwd&${pieceIds.join("&")}`;
+
+  visitUrl(url, true);
+  visitUrl("main.php");
+
+  const expected = pieces.join(",");
+
+  if (expected !== get("trainsetConfiguration", "")) {
+    throw new Error(
+      `Expected trainset configuration to have changed, expected "${expected}" but instead got ${get(
+        "trainsetConfiguration",
+        ""
+      )}`
+    );
+  }
+}
+
+class GarboWorkshed {
+  workshed: Item;
+  done: () => boolean;
+  action: () => void;
+
+  constructor(workshed: Item, done: () => boolean, action: () => void) {
+    this.workshed = workshed;
+    this.done = done;
+    this.action = action;
+  }
+}
+
+export function currentWorkshed(): GarboWorkshed {
+  return worksheds.find((workshed) => workshed.workshed === getWorkshed()) ?? defaultWorkshed;
+}
+
+export function getGarboWorkshed(item: Item): GarboWorkshed {
+  return worksheds.find((workshed) => workshed.workshed === item) ?? defaultWorkshed;
+}
+
+const trainset = new GarboWorkshed(
+  // eslint-disable-next-line libram/verify-constants
+  $item`model train set`,
+  () => {
+    // We should always get value from the trainset, so we would never switch from it
+    return false;
+  },
+  () => {
+    const pieces = [
+      TrainsetPiece.DOUBLE_NEXT_STATION,
+      TrainsetPiece.GAIN_MEAT,
+      TrainsetPiece.CANDY,
+      TrainsetPiece.RANDOM_BOOZE,
+      TrainsetPiece.DROP_LAST_FOOD_OR_RANDOM,
+      TrainsetPiece.ORE,
+      TrainsetPiece.EFFECT_MP,
+      TrainsetPiece.GAIN_STATS,
+    ];
+    if (!isTrainsetConfigurable()) return;
+    else if (pieces.join(",") === get("trainsetConfiguration", "")) return;
+    setTrainsetConfiguration(pieces);
+  }
+);
+
+const CMC = new GarboWorkshed(
+  $item`cold medicine cabinet`,
+  () => {
+    return get("_coldMedicineConsults") >= 5;
+  },
+  () => {
+    if (property.getNumber("_nextColdMedicineConsult") > totalTurnsPlayed()) return;
```

Comment:
This is unrelated to your PR except you are using worksheds - It would be good to update this code to not hit the CMC every turn and instead use the mafia properties to check

Reply:
I think this is out of scope for the current PR (I basically copied the existing `coldMedicineCabinet()` code).
The current code hits the CMC to check what items are available, then selects the best amongst them (since we don't want to choose an item we cannot currently select). 
I guess we can handle it similarly by parsing the most common character in `lastCombatEnvironment` and mapping it to its corresponding items to generate the same list of available items (only if "?" is not present in the preference), but this also seems like it would be more useful as a libram thing than a garbo thing as well (much like the trainset)